src/Density/Density.jl

```diff
@@ -264,6 +268,156 @@ function show(io::IO, g::MeltDependent_Density)
+
+Defines the BubbleFlow_Density as described in Slezin (2003) with a default gas solubility constant of 0.0041MPa``^{-1/2}`` used in e.g. Sparks et al. (1978)
+```math
+    \\rho = \\frac{1}{\\frac{c_0 - c}{\\rho_g} + \\frac{1-(c_0-c)}{\\rho_m}}
+```
+with
+```math
+c =
+\\begin{cases}
+   aP^{1/2} & \\text{for } P < \\frac{c_0^2}{a^2} \\\\
+    c_0 & \\text{for } P \\geq \\frac{c_0^2}{a^2}
+\\end{cases}
+```
+# Arguments
+- `ρmelt`: Density of the melt
+- `ρgas`: Density of the gas
+- `c0`: Total volatile content
+- `a`: Gas solubility constant (default: 0.0041MPa``^{-1/2}``) (after Sparks et al., 1978)
+
+Possible values for a are 0.0032-0.0064MPa``^{-1/2}`` where the lower value corresponds to mafic magmas at rather large pressures (400-600MPa) and the higher value to felsic magmas at low pressures (0 to 100-200MPa) (after Slezin (2003))
+
+# Example
+```julia
+rheology = SetMaterialParams(;
+                      Phase=1,
+                      CreepLaws=(PowerlawViscous(), LinearViscous(; η=1e21Pa * s)),
+                      Gravity=ConstantGravity(; g=9.81.0m / s^2),
+                      Density= BubbleFlow_Density(ρmelt=ConstantDensity(ρ=2900kg/m^3), ρgas=ConstantDensity(ρ=1kg/m^3), c0=0.0, a=0.0041MPa^-1//2),
+                      )
+```
+
+# References
+- Slezin, Yu. B. (2003), The mechanism of volcanic eruptions (a steady state approach), Journal of Volcanology and Geothermal Research, 122, 7-50, https://doi.org/10.1016/S0377-0273(02)00464-X
+- Sparks, R. S. J.(1978), The dynamics of bubble formation and growth in magmas: A review and analysis, Journal of Volcanology and Geothermal Research, 3, 1-37, https://doi.org/10.1016/0377-0273(78)90002-1
+"""
+@with_kw_noshow struct BubbleFlow_Density{_T, U1, U2, U3, S1<:AbstractDensity, S2 <:AbstractDensity} <: ConduitDensity{_T}
+    ρmelt::S1 = ConstantDensity(ρ=2200kg/m^3)   # density of the melt
+    ρgas::S2 = ConstantDensity(ρ=1kg/m^3)       # density of the gas
+    c0::GeoUnit{_T, U1} = 0e0 * NoUnits         # total volatile content
+    a::GeoUnit{_T, U2} = 0.0041MPa^(-1//2)      # gas solubility constant
+    ρ::GeoUnit{_T,U3} = 2900.0kg / m^3          # to keep track on whether this struct is dimensional or not
+end
+
+BubbleFlow_Density(args...) = BubbleFlow_Density(args[1], args[2], convert.(GeoUnit, args[3:end])...)
+isdimensional(s::BubbleFlow_Density) = isdimensional(s.ρmelt)
+
+function param_info(s::BubbleFlow_Density) # info about the struct
+    return MaterialParamsInfo(; Equation=L"\rho = 1/((c_0-c)/rho_g + 1-(c_0-c)/\rho_m)")
+end
+
+# Calculation routines
+@inline function (rho::BubbleFlow_Density{_T})(; P = 0e0, kwargs...) where {_T}
+    ρmelt = compute_density(rho.ρmelt, kwargs)
+    ρgas  = compute_density(rho.ρgas,  kwargs)
+    if P isa Quantity
+        @unpack_units c0, a = rho
+    else
+        @unpack_val c0, a = rho
+    end
+
+    cutoff = c0^2/a^2
+
+    if P < cutoff
+        c = a * P^(1//2)
```

Comment:
```suggestion
        c = a * sqrt(P)
```
This should be just the square root of `P`. It is much more efficient than the power of a rational or floating point number:
```julia-repl
julia> x=rand()
0.2628663876104308

julia> @b $x^(1//2)
13.816 ns

julia> @b $x^(1/2)
12.526 ns

julia> @b sqrt($x)
1.992 ns```